### PR TITLE
extend debug page to show assigned domain quota

### PR DESCRIPTION
**Pull request by heavygale**
This PR extends the disk usage display on the admin's debug page, adding a second progress bar for the total amount of assigned domain quota. This way, the admin can see how much of the available disk space has already been assined to domains, since this disk space must be considered as potentially used. The code can handle disk space overcommitmend, as the progressbar is capped to 100%.

before:
old
after:
new

I changed the dockerapi container to not return human readable values for df, so that the values can be used for calculations. Looks like "df" from dockerapi isn't used elsewhere in mailcow.

### CODE REVIEW

**Comment by MAGICCC** [comment]
Neat, possible to change the color of the bar(s) if you hit a certain threshold of like 75%/90% disk usage?

**Comment by andryyy** [comment]
What if you assigned more than 100%?

**Comment by heavygale** [comment]
> The code can handle disk space overcommitmend, as the progressbar is capped to 100%.

File: data/web/debug.php [hunk 1]
```diff
@@ -43,6 +43,11 @@
         <?php
           $exec_fields = array('cmd' => 'system', 'task' => 'df', 'dir' => '/var/vmail');
           $vmail_df = explode(',', json_decode(docker('post', 'dovecot-mailcow', 'exec', $exec_fields), true));
+          $used_percent = substr($vmail_df[4], 0, -1);
+          $quota_stats = mailbox('get','quota_stats');
+          $quotaPercent1 = round(($quota_stats['total_quota']/$vmail_df[3])*100);
+          $quotaPercent2 = round((($quota_stats['total_quota']-$quota_stats['used_bytes'])/$vmail_df[3])*100);
+          $quotaPercent2 = ($quotaPercent2+$used_percent>100) ? 100-$used_percent : $quotaPercent2;
```
Comment (heavygale) [inline data/web/debug.php:50]: Handling overcommitment -> progressbar is capped to 100%

Comment (andryyy) [reply data/web/debug.php:50]: Mörciii

**Comment by andryyy** [comment]
image

Not really working. Can you fix it?

File: data/Dockerfiles/dockerapi/server.py [hunk 2]
```diff
@@ -206,7 +206,7 @@ def container_post__exec__system__fts_rescan(self, container_id):
   def container_post__exec__system__df(self, container_id):
     if 'dir' in request.json:
       for container in docker_client.containers.list(filters={"id": container_id}):
-        df_return = container.exec_run(["/bin/bash", "-c", "/bin/df -H '" + request.json['dir'].replace("'", "'\\''") + "' | /usr/bin/tail -n1 | /usr/bin/tr -s [:blank:] | /usr/bin/tr ' ' ','"], user='nobody')
+        df_return = container.exec_run(["/bin/bash", "-c", "/bin/df '" + request.json['dir'].replace("'", "'\\''") + "' | /usr/bin/tail -n1 | /usr/bin/tr -s [:blank:] | /usr/bin/tr ' ' ','"], user='nobody')
```
Comment (heavygale) [inline data/Dockerfiles/dockerapi/server.py:209]: @andryyy We need a new build of dockerapi, as I removed the -H.

Comment (andryyy) [reply data/Dockerfiles/dockerapi/server.py:209]: I did locally, but the view is broken anyway. 

My VM has 100G, the other stuff is wrong, too.

**Comment by andryyy** [comment]
Plus:

image

**Comment by MAGICCC** [comment]
@andryyy Possible to revert? We discuss some stuff in IRC

**Pull request merged**